### PR TITLE
View results

### PR DESCRIPTION
**Pull request by PhiLhoSoft**
Output the generated HTML with CSS to files to allow to inspect the results.
Also fixes some style issues and makes the styles more consistent.

### CODE REVIEW

File: src/classes-overload-test/render.js [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
 
-export const renderHtml = (containerClassName, getButtonClassName) =>
-    `<section class=${containerClassName}>
-        ${(() => {
-            const buttons = [];
+export const renderHtml = (css, body) =>
+`<html>
+    <head>
+        <style type="text/css">
+${css}
```
Comment (PepijnSenders) [inline src/classes-overload-test/render.js]: Why don't you `tab` these? Looks better on the view?

File: src/run.js [hunk 2]
```diff
@@ -35,3 +34,24 @@ export default (testName, cases) => {
...
+    console.log(`Running view ${testName}.\n`);
+
+    const outputDir = createOutputDir(testName);
+
+    Object.keys(testCases).forEach((caseName) => {
+        const html = testCases[caseName].testCase(pad(caseName));
+        fs.writeFile(`${outputDir}/${caseName}.html`, html, (err) => {
+            if (err) {
+                console.log(err);
+            }
```
Comment (PepijnSenders) [inline src/run.js]: Maybe also output it to the console (output + congrats you're done notification).

File: src/run.js [hunk 3]
```diff
@@ -35,3 +34,24 @@ export default (testName, cases) => {
 
     return testSuite.run({ async: true });
```
Comment (PepijnSenders) [inline src/run.js]: Actually I would like both run methods to return a Promise when they're done.

**Comment by PepijnSenders** [comment]
Hey @PhiLhoSoft, thanks again for the update, looks really cool! Just gave you some comments.

File: src/run.js (hunk 3 above)
Comment (PhiLhoSoft) [inline src/run.js]: I haven't used much promises in Node.js, I can give them a try. Maybe later?

**Comment by PhiLhoSoft** [comment]
Not sure why there is a conflict, I rebased my branch on master. GitHub's *Resolve conflicts* button leads to a page that never finishes to load.

File: src/classes-overload-test/render.js (hunk 1 above)
Comment (PhiLhoSoft) [inline src/classes-overload-test/render.js]: Yes, some people (including me) prefer to inspect nice code... 😄 
Feel free to fix that if you prefer.
That's the problem of most multiline string implementations: you have to choose between code indentation and proper string content... Some languages (Ceylon) take this in account.

Comment (PepijnSenders) [inline src/classes-overload-test/render.js]: 👍

**Comment by PepijnSenders** [comment]
Yeah, that's weird looks like a bug in GItHub.

File: src/run.js [hunk 4]
```diff
@@ -1,37 +1,76 @@
...
+
+        const outputDir = createOutputDir(testName.replace(/ /, '-'));
+
+        Object.keys(testCases).forEach((caseName) => {
+            const html = testCases[caseName].testCase(pad(caseName));
+            const path = `${outputDir}/${caseName}.html`;
+            fs.writeFile(path, html, (err) => {
+                if (err) {
+                    console.error(err);
+                    reject();
```
Comment (PepijnSenders) [inline src/run.js]: You can reject the `err` here.

File: src/run.js [hunk 5]
```diff
@@ -1,37 +1,76 @@
...
+                } else {
+                    console.log(`Wrote ${path}`);
+                    resolve();
```
Comment (PepijnSenders) [inline src/run.js]: You may return the path here.

File: src/run.js [hunk 6]
```diff
@@ -1,37 +1,76 @@
...
+        });
 
-    return testSuite.run({ async: true });
+        console.log(`Running view ${testName}.\n`);
+
+        const outputDir = createOutputDir(testName.replace(/ /, '-'));
+
+        Object.keys(testCases).forEach((caseName) => {
+            const html = testCases[caseName].testCase(pad(caseName));
+            const path = `${outputDir}/${caseName}.html`;
```
Comment (PepijnSenders) [inline src/run.js]: We also should put this path in the gitignore.

**Comment by PepijnSenders** [comment]
LGTM @PhiLhoSoft, thanks again for your contributions!

**Comment by PhiLhoSoft** [comment]
Thanks for your guidance! 😆

**Pull request merged**